#43 user argumnet resolver 구현

Jennorresothie opened the pull request:
- 어노테이션 생성 UserInfo

- User정보 담는 DTO생성 UserInfo

- setArrtibute UserInfo 객체로 저장
- WebCongfig 에 Resolver 추가
- 각 Controller userId 부분 코드 변경
- UserNotFoundException 기본 생성자 추가

close #43

Jennorresothie committed:
#43 user argumnet resolver 구현

- 어노테이션 생성
UserInfo

- User정보 담는 DTO생성
UserInfo

- setArrtibute UserInfo 객체로 저장
- WebCongfig 에 Resolver 추가
- 각 Controller userId 부분 코드 변경
- UserNotFoundException 기본 생성자 추가

jgm0327 reviewed (approved):
👍

jhchoi57 commented on src/main/java/com/jootcamp/superboard/common/filter/LoginCheckFilter.java:
```diff
@@ -18,7 +18,7 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
                 System.out.println("통과");
                 HttpSession session = request.getSession(false);
-                if (session == null || session.getAttribute("userId") == null) {
+                if (session == null || session.getAttribute("userInfo") == null) {
```
common에 Constants 파일 만들고"userInfo" Constants 로 빼주세요

jhchoi57 commented on src/main/java/com/jootcamp/superboard/user/repository/exception/UserNotFoundException.java:
```diff
@@ -6,4 +6,7 @@ public class UserNotFoundException extends NotFoundException {
         super("해당하는 이메일을 찾을 수 없습니다. "+ email);
     }
+    public UserNotFoundException () {
+        super("해당하는 이메일을 찾을 수 없습니다. ");
```
클래스명은 UserNotFoundException 인데 메세지는 이메일을 찾을 수 없습니다 라서 조금 이상하네요 <br>
UserEmailNotFoundException 은 어떠신지

Jennorresothie committed:
Refactor(BE) : #43 피드백 반영한 수정

1. "userInfo" Constants 로 사용
변경 내역
- UserConstant
- LoginCheckFilter
- UserArgumentResolver
- UserService

2. UserNotFoundException -> UserEmailNotFoundException 로 클래스 명 변경

Jennorresothie committed:
Revert "Refactor(BE) : #43 피드백 반영한 수정"

This reverts commit 315f6d80abbff9b9b728ac52fef6398799474581.

Jennorresothie committed:
Reapply "Refactor(BE) : #43 피드백 반영한 수정"

This reverts commit 69936bbdbce82d8fd8b0a3ea86782b48ed8b980c.

Jennorresothie committed:
Auto stash before checking out "HEAD"
